lib/AppInfo/Application.php

```diff
@@ -57,18 +69,29 @@ public function onCreateOrUpdate(\OCP\Files\Node $node) {
 			/** @var $operation Operation */
 			$operation->considerConversion($node);
 		} catch (QueryException $e) {
-			$logger = $this->getContainer()->getServer()->getLogger();
-			$logger->logException($e, ['app' => 'workflow_pdf_converter', 'level' => ILogger::ERROR]);
+			$this->logger->logException($e, ['app' => 'workflow_pdf_converter', 'level' => ILogger::ERROR]);
+		}
+	}
+
+	public function onTagAssigned(MapperEvent $event) {
+		if($event->getObjectType() !== 'files') {
+			return;
+		}
+
+		$nodes = $this->rootFolder->getById((int)$event->getObjectId());
+		foreach($nodes as $node) {
```

Comment:
I would recommend `$node = array_shift($nodes)` instead?
Or do you want to crease 10 PRs if the user has 10 ways to access a file?

Reply:
It's sort of arbitrary, but doing the job multiple times does not make sense indeed.